openrtx/include/fonts/symbols/scripts/generate_symbols.py

```diff
@@ -0,0 +1,158 @@
+    scalar = 1.6  # this number is used to roughly convert from pt to px
+    yadvance_scalar = (
+        7 / 3
+    )  # this number is applied to the width to scale up the yadvance; this is naively linear
+
+    def __init__(self, name, size):
+        # Prepopulate with a space character
+        self.bitmaps = [[0x00]]
+        self.glyphNames = ["space"]
+        self.initial_offset = 1
+
+        self.size = size
+        self.name = name
+        self.height = round(size * self.scalar)
+        self.width = round(size * self.scalar)
+        self.yadvance = round(size * self.scalar * self.yadvance_scalar)
+        self.xadvance = round(size * self.scalar * 2 / 3)
+        self.dY = round(-1 * self.height / 1.2)
+
+    def addGlyph(self, filename, glyphName):
+        res = svgToBytes(filename, self.height)
+        packed = numpy.packbits(res)
+        charBytes = packed.tobytes()
+
+        self.bitmaps.append(charBytes)
+
+        self.glyphNames.append(glyphName)
+
+    def __str__(self):
+        out = "static const uint8_t {}{}pt7bBitmaps[] PROGMEM = {{\n".format(
+            self.name, self.size
+        )
+        for i, bitmap in enumerate(self.bitmaps):
+            out += (
+                (", ".join("0x{:02x}".format(x) for x in bitmap))
+                + ", //"
+                + self.glyphNames[i]
+                + "\n"
+            )
+        out += "};\n"
+        out += "\n"
+        out += (
+            "static const GFXglyph {}{}pt7bGlyphs[] PROGMEM = {{".format(
+                self.name, self.size
+            )
+            + "\n"
+        )
+        out += "  //Index,  W, H,xAdv,dX, dY\n"
+        offset = 0
+        for glyphNumber, glyphName in enumerate(self.glyphNames):
+            out += '{{ {}, {}, {}, {}, 0, {}}},   // "{}" {}\n'.format(
+                offset,
+                self.width if glyphNumber > 0 else 1,
+                self.height if glyphNumber > 0 else 1,
+                self.xadvance,
+                self.dY,
+                chr(self.first_ascii + glyphNumber),
+                glyphName,
+            )
+            offset += len(self.bitmaps[glyphNumber])
+        out += "};\n"
+        out += (
+            "static const GFXfont {}{}pt7b PROGMEM = {{".format(self.name, self.size)
+            + "\n"
+        )
+        out += "(uint8_t  *){}{}pt7bBitmaps,".format(self.name, self.size) + "\n"
+        out += "(GFXglyph *){}{}pt7bGlyphs,".format(self.name, self.size) + "\n"
+        out += "//ASCII start, ASCII stop,y Advance \n"
+        out += (
+            "  {}, {}, {} }};".format(
+                self.first_ascii,
+                self.first_ascii + len(self.glyphNames) - 1,
+                round(self.yadvance),
+            )
+            + "\n"
+        )
+        return out
+
+
+###################################################################
+
+for size in sizes:
+    font = FontDefinition("Symbols", size)
+    for symbol in symbols:
```

Comment:
Do we want to hardcode all svgs or use something like `for file in glob.glob(r'../sources/*.svg'):`?
This would make it easier to add svgs, by not requiring to touch the code when adding new icons

Reply:
The downside here is that the order is what determines the index. So unless we're strict about the file names, it risks a new file being added in a location that would shift the chars used to access existing symbols.

Reply:
Ok, so it will be a tradeoff between maintaining the list or maintaining a coherent naming scheme for the svgs.
I think the list will then be easier to maintain.

Reply:
Good news, by adding an enum for the symbol name and having that autogenerated too, now this is indeed auto-discovered rather than hand curated.